### PR TITLE
Improve request tracking error handling

### LINKED ISSUE #194: Tracking requests are stored but none are processed
I have QueuedTracking with Redis and using Memurai on a Windows machine. When plugin is active, I can see items in queue but none of them are processed. 

Exception from the matomo.log:
```
#0\core\Tracker\RequestSet.php(161),
#1\plugins\QueuedTracking\Queue.php(94),
#2\plugins\QueuedTracking\Queue\Processor.php(112),
#3\plugins\QueuedTracking\Commands\Process.php(86),
#4\vendor\symfony\console\Symfony\Component\Console\Command\Command.php(257),
#5\vendor\symfony\console\Symfony\Component\Console\Application.php(874),
#6\vendor\symfony\console\Symfony\Component\Console\Application.php(195),
[internal function]: Symfony\Component\Console\Application->doRun(),
#8\core\Console.php(135),
#9\core\Access.php(670)
```

Have tried processing the queue using the "Process during tracking request" and manually via `/console queuedtracking:process`.

Example output when monitoring:
```
35 (19+16) request sets left in queue. 810.84K used memory (1020.51K peak). 0 workers active.
35 (19+16) request sets left in queue. 810.84K used memory (1020.51K peak). 0 workers active.
36 (19+17) request sets left in queue. 810.84K used memory (1020.51K peak). 0 workers active.
```

Output when processing manually:
```
Starting to process request sets, this can take a while

This worker finished queue processing with 0req/s (0 requests in 0.11 seconds)
```

I am using these versions:
PHP: 8.0
Matomo: 4.11.0
MariaDB: 10.5.9.0
Memurai: 3.1.4
PHPRedis: 5.3.7

Output from `/console queuedtracking:test`
```
Settings that will be used:
Backend: redis
NumQueueWorkers: 2
NumRequestsToProcess: 5
ProcessDuringTrackingRequest: 1
QueueEnabled: 1

Redis backend only settings (does not apply when using MySQL backend):
Host: *******
Port: 6379
Timeout: 0
Password:
Database: 0
UseSentinelBackend: 0
SentinelMasterName: mymaster

Version / stats:
PHP version: 8.0.0
Uname: Windows NT DMZWA02 10.0 build 17763 (Windows Server 2016) AMD64
PHPRedis version: 5.3.7
Backend version: 6.2.6
Memory: array (
  'used_memory' => 914360,
  'used_memory_human' => '892.93K',
  'used_memory_rss' => 856472,
  'used_memory_rss_human' => '836.40K',
  'used_memory_peak' => 1045000,
  'used_memory_peak_human' => '1020.51K',
  'used_memory_peak_perc' => '87.50%',
  'used_memory_overhead' => 850600,
  'used_memory_startup' => 685912,
  'used_memory_dataset' => 63760,
  'used_memory_dataset_perc' => '27.91%',
  'allocator_allocated' => 957376,
  'allocator_active' => 1486848,
  'allocator_resident' => 818584,
  'total_system_memory' => 0,
  'total_system_memory_human' => '0B',
  'used_memory_lua' => 37888,
  'used_memory_lua_human' => '37.00K',
  'used_memory_scripts' => 416,
  'used_memory_scripts_human' => '416B',
  'number_of_cached_scripts' => 2,
  'maxmemory' => 0,
  'maxmemory_human' => '0B',
  'maxmemory_policy' => 'noeviction',
  'allocator_frag_ratio' => 1.55,
  'allocator_frag_bytes' => 529472,
  'allocator_rss_ratio' => 0.55,
  'allocator_rss_bytes' => -668264,
  'rss_overhead_ratio' => 1.05,
  'rss_overhead_bytes' => 37888,
  'mem_fragmentation_ratio' => 1.0,
  'mem_fragmentation_bytes' => 0,
  'mem_not_counted_for_evict' => 0,
  'mem_replication_backlog' => 0,
  'mem_clients_slaves' => 0,
  'mem_clients_normal' => 164128,
  'mem_aof_buffer' => 0,
  'mem_allocator' => 'jemalloc-3.6.0',
  'active_defrag_running' => 0,
  'lazyfree_pending_objects' => 0,
  'lazyfreed_objects' => 0,
)
MaxMemory Eviction Policy config: noeviction
MaxMemory config: 0

Performing some tests:
Redis is connected: 1
Connection works in general
Success for method set(testKey, value)
Success for method setnx(testnxkey, value)
Success for method setex(testexkey, 5, value)
Success for method set(testKeyWithNx, value, array (0 => 'nx',))
Success for method set(testKeyWithEx, value, array ('ex' => 5,))
Initial expire seems to be set correctly
setIfNotExists works fine
expireIfKeyHasValue seems to work fine
Extending expire seems to be set correctly
expireIfKeyHasValue correctly expires only when the value is correct
Expire is still set which is correct
deleteIfKeyHasValue seems to work fine
List feature seems to work fine
```

### PR DESCRIPTION
**Pull request by snake14**
### Description:

Some errors weren't getting caught properly and were preventing the queue from processing all the other requests. This improvement should resolve that issue.
Fixes: #194 

### Review

* [ ] [Functional review done](https://developer.matomo.org/guides/pull-request-reviews#functional-review-done)
* [ ] [Potential edge cases thought about](https://developer.matomo.org/guides/pull-request-reviews#potential-edge-cases-thought-about) (behavior of the code with strange input, with strange internal state or possible interactions with other Matomo subsystems)
* [ ] [Usability review done](https://developer.matomo.org/guides/pull-request-reviews#usability-review-done) (is anything maybe unclear or think about anything that would cause people to reach out to support)
* [ ] [Security review done](https://developer.matomo.org/guides/security-in-piwik#checklist)
* [ ] [Wording review done](https://developer.matomo.org/guides/pull-request-reviews#translations-wording-review-done)
* [ ] [Code review done](https://developer.matomo.org/guides/pull-request-reviews#code-review-done)
* [ ] [Tests were added if useful/possible](https://developer.matomo.org/guides/pull-request-reviews#tests-were-added-if-usefulpossible)
* [ ] [Reviewed for breaking changes](https://developer.matomo.org/guides/pull-request-reviews#reviewed-for-breaking-changes)
* [ ] [Developer changelog updated if needed](https://developer.matomo.org/guides/pull-request-reviews#developer-changelog-updated-if-needed)
* [ ] [Documentation added if needed](https://developer.matomo.org/guides/pull-request-reviews#documentation-added-if-needed)
* [ ] Existing documentation updated if needed

### CODE REVIEW

**Comment by snake14** [comment]
Ah. I forgot that uadata only became an issue in more recent releases. I'll have to come up with a different field that throws a TypeError in older Matomo versions as well.

File: tests/Integration/Queue/ProcessorTest.php [hunk 1]
```diff
@@ -204,6 +205,33 @@ public function test_processRequestSets_ShouldReturnOnlyValidRequestSetsInCaseTh
...
+        // The uadata field is expected to be a JSON string and not an array. It will throw a TypeError during decoding
+        $requestParams['uadata'] = [];
+        $requestSet2->setRequests([$requestSet2->getRequests()[0], new Tracker\Request($requestParams)]);
+
+        $this->acquireAllQueueLocks();
+        $requestSetsToRetry = $this->processor->processRequestSets($tracker, $queuedRequestSets);
+
+        // I couldn't find a param other than uadata that could throw a TypeError and it's not used in older versions
+        $isOlderMatomo = version_compare(Version::VERSION, '4.12.0', '>');
+        $expectedSets = $isOlderMatomo ? [$requestSet1, $requestSet2] : [];
```
Comment (AltamashShaikh) [inline tests/Integration/Queue/ProcessorTest.php]: @snake14 if you extend this test to re run, $expectedSets should be [$requestSet1] for newer version right ? If yes can you include that here too ?

**Review by AltamashShaikh — COMMENTED** [review]
@snake14 Left 1 comment for testcase

File: tests/Integration/Queue/ProcessorTest.php (hunk 1 above)
Comment (snake14) [reply tests/Integration/Queue/ProcessorTest.php]: Thank you @AltamashShaikh . Making that change made me realise that I misnamed the variable.

File: Configuration.php [hunk 2]
```diff
@@ -16,6 +16,8 @@ class Configuration
     const DEFAULT_NOTIFY_THRESHOLD = 250000;
     const KEY_NOTIFY_EMAILS = 'notify_queue_threshold_emails';
     const KEY_NOTIFY_THRESHOLD = 'notify_queue_threshold_single_queue';
+    const LOG_FAILED_TRACKING_REQUESTS = 'log_failed_tracking_request_body';
```
Comment (AltamashShaikh) [inline Configuration.php:19]: should we create an FAQ on where and how to use this ?

Comment (snake14) [reply Configuration.php:19]: Yep. I [updated the FAQ file](https://github.com/matomo-org/plugin-QueuedTracking/pull/214/commits/4ea3e8bbc11844ffb60bf6ceee8379bc2820d616) that's part of this repo.

Comment (AltamashShaikh) [reply Configuration.php:19]: Also should we add this setting in install ? Something like [this](https://github.com/matomo-org/plugin-QueuedTracking/blob/4.x-dev/Configuration.php#L33..L35)

**Review by AltamashShaikh — COMMENTED** [review]
@snake14 left 1 comment, rest looks good

File: Configuration.php (hunk 2 above)
Comment (snake14) [reply Configuration.php:19]: I didn't see any need to add it at install as customers won't use it unless they're debugging.

**Review by AltamashShaikh — APPROVED** [review]

**Pull request merged**